File: MyDent.Services/Database/DoctorAbsence.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyDent.Services.Database
{
    public class DoctorAbsence
    {
        [Key]
        public int Id { get; set; }

        public int DoctorId { get; set; }

        [ForeignKey("DoctorId")]
        public Doctor Doctor { get; set; } = null!;

        [Required]
        public DateOnly StartDate { get; set; }

        [Required]
        public DateOnly EndDate { get; set; }

        [Required]
        [MaxLength(300)]
        public string Reason { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: UI/mydent_desktop/lib/models/doctor_absence.dart
import 'package:json_annotation/json_annotation.dart';

part 'doctor_absence.g.dart';

@JsonSerializable()
class DoctorAbsence {
  final int? id;
  final int? doctorId;
  final String? doctorName;

  // DateOnly serializes as "yyyy-MM-dd" — kept as raw string, parsed where needed.
  final String? startDate;
  final String? endDate;
  final String? reason;
  final DateTime? createdAt;

  DoctorAbsence({
    this.id,
    this.doctorId,
    this.doctorName,
    this.startDate,
    this.endDate,
    this.reason,
    this.createdAt,
  });

  factory DoctorAbsence.fromJson(Map<String, dynamic> json) =>
      _$DoctorAbsenceFromJson(json);

  Map<String, dynamic> toJson() => _$DoctorAbsenceToJson(this);
}
